dev/devicelab/lib/framework/adb.dart

```diff
@@ -551,6 +618,20 @@ class IosDeviceDiscovery implements DeviceDiscovery {
     print('Device chosen: $_workingDevice');
   }
 
+  @override
+  Future<void> chooseWorkingDeviceById(String deviceId) async {
+    deviceId = _findMatchId(await discoverDevices(), deviceId);
+    if (deviceId != null) {
+      _workingDevice = IosDevice(deviceId: deviceId);
+      print('Choose device by ID: ' + _workingDevice.toString());
+      return;
+    }
+    throw DeviceException(
+      'Device with ID $deviceId is not found for operating system: '
```

Comment:
Here and above, it looks like there is some shadowing over `deviceId`, this point would only be reached if it was null right? Or is the intention to print the original `deviceId` passed as an argument?

Reply:
Thanks for the catch! It is a bad design to log the unmatched null deviceId. Fixed.